proxy/core/src/main/scala/io/cloudstate/proxy/AkkaActorRegisterFeature.scala

```diff
@@ -8,26 +8,25 @@ import scala.collection.JavaConverters._
 
 @AutomaticFeature
 final class AkkaActorRegisterFeature extends Feature {
-
+  val akkaActorClass = classOf[akka.actor.Actor]
   override def duringAnalysis(access: Feature.DuringAnalysisAccess) =
-    reachableSubtypes(access, access.findClassByName("akka.akka.Actor")).foreach { cls =>
-      for {
-        context <- getDeclaredField(cls, "context")
-        self <- getDeclaredField(cls, "self")
-      } {
-        RuntimeReflection.register(cls)
-        RuntimeReflection.register( /* finalIsWritable = */ true, context, self)
-        RuntimeReflection.register(cls.getDeclaredFields: _*)
-        RuntimeReflection.register(cls.getDeclaredConstructors: _*)
+    if (access.isReachable(akkaActorClass)) {
+      access.reachableSubtypes(akkaActorClass).iterator.asScala.filter(_ != null).foreach { subtype =>
+        if (subtype.getInterfaces.exists(_ == akkaActorClass)) {
```

Comment:
Why this?  What's the benefit and/or need for it?

Reply:
Because the *fields* only exist on the class which mixes in the Actor trait.

Reply:
I see.  So that makes sense, but does it mean that we might not be properly registering the constructors of other subtypes of Actor?  Sounds like a potential problem.

Reply:
@dwijnand That's a good point. We might want to separate those registrations from one another.